---
title: "Infections à *Mycobacterium abscessus* : épidémiologie et chimio-résistance"
author: "Vichita Ok"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
    html_document:
      self_contained: true
      number_sections: false
      code_folding: "hide"
      toc: true
      toc_depth: 3
      toc_float: true
---


```{r settings, include=FALSE}
knitr::opts_chunk$set(eval= FALSE, echo =TRUE, cache = FALSE, message = FALSE, warning = FALSE, cache.lazy = FALSE,
                      fig.height = 3.5, fig.width = 10.5)
```


# Introduction

## *Mycobacterium abscessus*

* Characteristics

    + Rapid growing mycobacterium 
    + Environmental contaminant  


* Subspecies of M. abscessus

    + *M. abscessus senso strictu*
    + *M. massiliense*
    + *M. bolletii*  


* Pathogenicity

    + Pulmonary disease ++
    + Cutaneous disease
    + Disseminated disease: rare (ID)  
    
* Chemoresistance

    + Intrinsic resistance
        - rifampicin
        - tobramycin
        - ethambutol
        - fluoroquinolones  
        
    + Acquired resistance
        - Macrolides (mutations *rrl*)
        - Aminosides (mutations *rrs*)


# Analyse

## Construction du jeu de données

### Données publiques NCBI

**Liste des génomes**

[liste](https://www.ncbi.nlm.nih.gov/genome/browse/#!/prokaryotes/mycobacterium%20abscessus%20subsp.%20abscessus) (*Mycobacteroides abscessus subsp. abscessus*): 


**Script pour le téléchargement des génomes**

[script](https://github.com/vichitaok/projet_tutore/blob/main/scripts/ncbi_fna.sh).

Le script a été testé sur 14 génomes et a bien fonctionné.
```{bash test telechargement genomes}
#script testé sur 14 genomes "complete" : fonctionné
qsub -cwd -V -S -N ncbi-genomes ncbi_fna.sh 1>output.log 2>err.log
```

912 genomes (= 922 - 10 (pour lesquels aucun lien ftp)) ont été téléchargés à l'aide du [script](https://github.com/vichitaok/projet_tutore/blob/main/scripts/ncbi_fna.sh).

Le script a été modifié pour le téléchargement des 912 génomes :
- présence de côtes "" dans le tableau

```{bash telechargement genomes}
# soumission du script pour telecharger les genomes
qsub -cwd -V -N ncbi_genomes -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q /save_projet/mab_resistance/projet_tutore/scripts/ncbi_fna.sh

# extraction des fichiers .fna.gz
qsub -cwd -V -N extract -b y "gunzip /work_projet/mab_resistance/GENOMES_NCBI/*.fna.gz"

```

**Génomique comparative** 

sur les 912 génomes téléchargés avec **dRep**.

```{bash dRep comparaison genomes ncbi}
qsub -cwd -V -N drep -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi -g /work_projet/mab_resistance/DATA/GENOMES_NCBI/*.fna && conda deactivate"

```

Le job a été interrompu car a dépassé les 5 jours de la queue long.
Pour réduire le temps de calcul, ont été ajoutées les options :

- `--greedy_secondary_clustering` : 
Use a heuristic to avoid pair-wise comparisons when doing secondary clustering. Will be done with single linkage clustering. Only works for fastANI S_algorithm option at the moment ;
- `--S_algorithm fastANI` : Kmer-based approach; very fast.

```{bash dRep comparaison genomes ncbi greedy_secondary_clustering + fast_ANI}
qsub -cwd -V -N drep -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep dereplicate -d -pa 0.99 -sa 0.999 -p 32 --S_algorithm fastANI --greedy_secondary_clustering /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi -g /work_projet/mab_resistance/DATA/GENOMES_NCBI/*.fna && conda deactivate"

#$ New checkM db needs to be made
```

Mais erreur `New checkM db needs to be made`, malgré l'utilisation les 200 Go dispos sur `\tmp` du noeud long.
checkM nécessite un espace considérable. 

En parcourant les liens suivants:
[https://drep.readthedocs.io/en/latest/advanced_use.html#troubleshooting-checkm](https://drep.readthedocs.io/en/latest/advanced_use.html#troubleshooting-checkm) et [https://github.com/MrOlm/drep/issues/45](https://github.com/MrOlm/drep/issues/45), 

l'idée est de créer un nouveau répertoire `\tmp` dans le `/work_projet/mab_resistance/` pour lancer checkM séparément et passer les résultats obtenus à dRep.

*checkM*

- checkm `lineage_wf`
- checkm `qa`

```{bash checkM lineage_wf}
qsub -cwd -V -N checkm -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && checkm lineage_wf /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/prodigal/ /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/ -f /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir//results.tsv --tab_table -t 32 --pplacer_threads 32 -g -x faa --tmpdir /work_projet/mab_resistance/tmp/ && conda deactivate"

```

```{bash checkM qa}
qsub -cwd -V -N checkm-qa -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && checkm qa /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/lineage.ms /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/ -f /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/Chdb.tsv -t 32 --tab_table -o 2 --tmpdir /work_projet/mab_resistance/tmp/ && conda deactivate"
```

*dRep*

- dRep avec option `--genomeInfo` 
- sur le résultat de checkM (format .csv avec les colonnes `genome,completeness,contamination`)

```{bash dRep}
## obtenir un fichier .csv avec les colonnes genome,completeness,contamination à passer sur dRep
cut -f 1,12,13 results.tsv | sed 's/Completeness/completeness/' | sed 's/Contamination/contamination/' | sed 's/Bin Id/genome/' | sed 's/\t/,/g' > genome_info.csv

## dREp avec option --genomeInfo
qsub -cwd -V -N drep -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep dereplicate --debug --genomeInfo /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/genome_info.csv -pa 0.99 -sa 0.999 -p 32 --S_algorithm fastANI --greedy_secondary_clustering /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi -g /work_projet/mab_resistance/DATA/GENOMES_NCBI/*.fna && conda deactivate"

```

**Résultats**

[dRep](https://github.com/vichitaok/projet_tutore/tree/main/results/ncbi_genomes/drep/figures)




### Séquences *M. abscessus* hôpital

* 81 séquences obtenues à partir de 81 souches isolées de prélèvements biologiques.
  
  - paired-end reads (2 x 150 pb)
  - Nextera XT DNA Library Preparation Kit / MiniSeq (300-cycles) et MiSeq 
  
  

* Contrôle qualité :

  [rapport MultiQC](https://vichitaok.github.io/projet_tutore/results/mab_hopital/multiqc_report.html)

```{bash fastqc}
#creation repertoires
for f in /work_projet/mab_resistance/DATA/MAB_HOPITAL/*.fastq.gz ; do mkdir -p /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/`basename $f .fastq.gz`/1-QC/fastqc/ ; done

#fastqc
for f in /work_projet/mab_resistance/DATA/MAB_HOPITAL/*.fastq.gz ; do 
  qsub -cwd -V -N fastqc-`basename ${f} .fastq.gz` 
  -q short.q -pe thread 4 
  -o /work_projet/mab_resistance/log/`basename ${f} .fastq.gz` 
  -e /work_projet/mab_resistance/log/`basename ${f} .fastq.gz` 
  -b y 
  "conda activate fastqc-0.11.8 && fastqc ${f} 
  --outdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/`basename ${f} .fastq.gz`/1-QC/fastqc/  
  && conda deactivate" ; 
done

#multiqc
mkdir -p /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/1-QC/multiqc/
mkdir -p /work_projet/mab_resistance/log/global/
qsub -cwd -V -N multiqc -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate multiqc-1.7 && multiqc -ip -o /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/1-QC/multiqc/ /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/1-QC/fastqc/ && conda deactivate"


```

* Nettoyage: outil fastp

```{bash fastp}
#liste des souches
ls /work_projet/mab_resistance/DATA/MAB_HOPITAL/ | sed 's/_R[12].fastq.gz//g' | uniq > /work_projet/mab_resistance/tables/mab_hopital_liste.txt

#fastp running
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do 
qsub -cwd -V -N fastp-${id} 
-pe thread 4 
-o /work_projet/mab_resistance/log/${id} 
-e /work_projet/mab_resistance/log/${id} 
-b y 
"conda activate fastp-0.20.0 && fastp 
--in1 /work_projet/mab_resistance/DATA/MAB_HOPITAL/${id}_R1.fastq.gz 
--in2 /work_projet/mab_resistance/DATA/MAB_HOPITAL/${id}_R1.fastq.gz 
--out1 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R1.cleaned_filtered.fastq.gz 
--out2 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R2.cleaned_filtered.fastq.gz 
--html /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_fastp.html 
--thread 4 
--detect_adapter_for_pe 
--cut_mean_quality 30 
--cut_window_size 8 
--length_required 100 
--cut_tail 
--json CLEANING/fastp.json && 
conda deactivate" ; 
done


#fastp en retirant les N bases
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do qsub -cwd -V -N fastp-${id} -pe thread 4 -o /work_projet/mab_resistance/log/${id} -e /work_projet/mab_resistance/log/${id} -b y "conda activate fastp-0.20.0 && fastp --in1 /work_projet/mab_resistance/DATA/MAB_HOPITAL/${id}_R1.fastq.gz --in2 /work_projet/mab_resistance/DATA/MAB_HOPITAL/${id}_R1.fastq.gz --out1 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R1.cleaned_filtered.fastq.gz --out2 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R2.cleaned_filtered.fastq.gz --html /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_fastp.html --thread 4 --detect_adapter_for_pe --cut_mean_quality 30 --cut_window_size 8 --length_required 100 --cut_tail --n_base_limit 0 --json CLEANING/fastp.json && conda deactivate" ; done


```


* Contrôle qualité après nettoyage:

  [rapport MultiQC après nettoyage](https://vichitaok.github.io/projet_tutore/results/mab_hopital/multiqc_on_cleaned_report.html)

```{bash}
#fastqc post-trimming
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do qsub -cwd -V -N fastqc-cleaned-${id} -q short.q -pe thread 4 -o /work_projet/mab_resistance/log/${id}/ -e /work_projet/mab_resistance/log/${id}/ -b y "conda activate fastqc-0.11.8 && fastqc /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/*.fastq.gz --outdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/1-QC/fastqc/  && conda deactivate" ; done


## multiqc global
# répertoire pour résultat
mkdir -p /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/1-QC/multiqc-cleaned-fastp/

# multiqc running
qsub -cwd -V -N multiqc -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate multiqc-1.7 && multiqc -ip -o /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/1-QC/multiqc-cleaned-fastp/ /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/1-QC/fastqc/ /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/2-CLEANING/fastp/ && conda deactivate"

```


[rapport global](https://vichitaok.github.io/projet_tutore/results/mab_hopital/multiqc_on_cleaned_fastp_report.html)


* Statistiques sur les données: 

```{bash seqkit stats}
qsub -cwd -V -N seqkit-stats -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate seqkit-0.11.0 && seqkit stats /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/2-CLEANING/fastp/*.cleaned_filtered.fastq.gz > /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/seqkit_stats_on_cleaned.txt && conda deactivate"

```

```{r libraries, eval=TRUE}
#### Required libraries ####

# Load required CRAN R libraries
required_cranLib <- c("knitr", 
                      "tidyverse")  # data manipulation
for (lib in required_cranLib) {
  if (!require(lib, character.only = TRUE)) {
    install.packages(lib)
  }
  require(lib, character.only = TRUE)
}

kable(as.data.frame(c(required_cranLib)),
      col.names = "libraries",
      caption = "Loaded required libraries"
    )
```


```{r calcul profondeurs, eval=TRUE}

reads_stat <- read.table("/work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/seqkit_stats_on_cleaned.txt",
                         header = TRUE)

# renommer les noms des reads (1ère colonne file)
reads_stat$file <- str_sub(reads_stat$file, 73, 80)

# transformer la colonne num_seqs (chr) en numeric pour calculs
reads_stat$num_seqs_num <- as.numeric(str_replace_all(reads_stat$num_seqs, ",", ""))

# calcul profondeur à partir des nb de sequences et longueur moyenne des reads
reads_stat <- reads_stat %>%
  mutate(depth = (as.numeric(num_seqs_num) * avg_len) / 2.5e6) # taille génome M.asbcessus ~ 5 Mbp

# view (reads_stat)
kable(reads_stat,
      caption = "Seqkit stats reads M.abscessus")

# summary
summary(reads_stat)
```

```{r}


```

* Assemblage : 

```{bash unicycler}
# Assembly
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do mkdir -p /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/ ; done

## Test sur un échantillon mab01
qsub -cwd -V -N unicycler -pe thread 4  -o /work_projet/mab_resistance/log/mab01/ -e /work_projet/mab_resistance/log/mab01/ -b y "conda activate unicycler-0.4.8 && unicycler  --short1 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/2-CLEANING/fastp/mab01_R1.cleaned_filtered.fastq.gz  --short2 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/2-CLEANING/fastp/mab01_R2.cleaned_filtered.fastq.gz   --out /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/3-Assembly/ --threads 4 && conda deactivate" 


```

test assemblage unicycler sur mabO1
81 contigs





