---
title: "Infections à *Mycobacterium abscessus* : épidémiologie et chimio-résistance"
author: "Vichita Ok"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
    html_document:
      self_contained: true
      number_sections: false
      code_folding: "hide"
      toc: true
      toc_depth: 3
      toc_float: true
---


```{r settings, include=FALSE}
knitr::opts_chunk$set(eval= FALSE, echo =TRUE, cache = FALSE, message = FALSE, warning = FALSE, cache.lazy = FALSE,
                      fig.height = 3.5, fig.width = 10.5)
```


# Introduction

## *Mycobacterium abscessus*

* Characteristics

    + Rapid growing mycobacterium 
    + Environmental contaminant  


* Subspecies of M. abscessus

    + *M. abscessus senso strictu*
    + *M. massiliense*
    + *M. bolletii*  


* Pathogenicity

    + Pulmonary disease ++
    + Cutaneous disease
    + Disseminated disease: rare (ID)  
    
* Chemoresistance

    + Intrinsic resistance
        - rifampicin
        - tobramycin
        - ethambutol
        - fluoroquinolones  
        
    + Acquired resistance
        - Macrolides (mutations *rrl*)
        - Aminosides (mutations *rrs*)


# Analyse

## Construction du jeu de données

### Données publiques NCBI

**Liste des génomes**

[liste](https://www.ncbi.nlm.nih.gov/genome/browse/#!/prokaryotes/mycobacterium%20abscessus%20subsp.%20abscessus) (*Mycobacteroides abscessus subsp. abscessus*): 


**Script pour le téléchargement des génomes**

[script](https://github.com/vichitaok/projet_tutore/blob/main/scripts/ncbi_fna.sh).

Le script a été testé sur 14 génomes et a bien fonctionné.
```{bash test telechargement genomes}
#script testé sur 14 genomes "complete" : fonctionné
qsub -cwd -V -S -N ncbi-genomes ncbi_fna.sh 1>output.log 2>err.log
```

912 genomes (= 922 - 10 (pour lesquels aucun lien ftp)) ont été téléchargés à l'aide du [script](https://github.com/vichitaok/projet_tutore/blob/main/scripts/ncbi_fna.sh).

Le script a été modifié pour le téléchargement des 912 génomes :
- présence de côtes "" dans le tableau

```{bash telechargement genomes}
# soumission du script pour telecharger les genomes
qsub -cwd -V -N ncbi_genomes -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q /save_projet/mab_resistance/projet_tutore/scripts/ncbi_fna.sh

# extraction des fichiers .fna.gz
qsub -cwd -V -N extract -b y "gunzip /work_projet/mab_resistance/GENOMES_NCBI/*.fna.gz"

```

## Génomique comparative sur les génomes téléchargés

sur les 912 génomes téléchargés avec **dRep**.

```{bash dRep comparaison genomes ncbi}
qsub -cwd -V -N drep -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi -g /work_projet/mab_resistance/DATA/GENOMES_NCBI/*.fna && conda deactivate"

```

Le job a été interrompu car a dépassé les 5 jours de la queue long.
Pour réduire le temps de calcul, ont été ajoutées les options :

- `--greedy_secondary_clustering` : 
Use a heuristic to avoid pair-wise comparisons when doing secondary clustering. Will be done with single linkage clustering. Only works for fastANI S_algorithm option at the moment ;
- `--S_algorithm fastANI` : Kmer-based approach; very fast.

```{bash dRep comparaison genomes ncbi greedy_secondary_clustering + fast_ANI}
qsub -cwd -V -N drep -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep dereplicate -d -pa 0.99 -sa 0.999 -p 32 --S_algorithm fastANI --greedy_secondary_clustering /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi -g /work_projet/mab_resistance/DATA/GENOMES_NCBI/*.fna && conda deactivate"

#$ New checkM db needs to be made
```

Mais erreur `New checkM db needs to be made`, malgré l'utilisation les 200 Go dispos sur `\tmp` du noeud long.
checkM nécessite un espace considérable. 

En parcourant les liens suivants:
[https://drep.readthedocs.io/en/latest/advanced_use.html#troubleshooting-checkm](https://drep.readthedocs.io/en/latest/advanced_use.html#troubleshooting-checkm) et [https://github.com/MrOlm/drep/issues/45](https://github.com/MrOlm/drep/issues/45), 

l'idée est de créer un nouveau répertoire `\tmp` dans le `/work_projet/mab_resistance/` pour lancer checkM séparément et passer les résultats obtenus à dRep.

*checkM*

- checkm `lineage_wf`
- checkm `qa`

```{bash checkM lineage_wf}
qsub -cwd -V -N checkm -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && checkm lineage_wf /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/prodigal/ /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/ -f /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir//results.tsv --tab_table -t 32 --pplacer_threads 32 -g -x faa --tmpdir /work_projet/mab_resistance/tmp/ && conda deactivate"

```

```{bash checkM qa}
qsub -cwd -V -N checkm-qa -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && checkm qa /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/lineage.ms /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/ -f /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/Chdb.tsv -t 32 --tab_table -o 2 --tmpdir /work_projet/mab_resistance/tmp/ && conda deactivate"
```

*dRep*

- dRep avec option `--genomeInfo` 
- sur le résultat de checkM (format .csv avec les colonnes `genome,completeness,contamination`)

```{bash dRep}
## obtenir un fichier .csv avec les colonnes genome,completeness,contamination à passer sur dRep
cut -f 1,12,13 results.tsv | sed 's/Completeness/completeness/' | sed 's/Contamination/contamination/' | sed 's/Bin Id/genome/' | sed 's/\t/,/g' > genome_info.csv

## dREp avec option --genomeInfo
qsub -cwd -V -N drep -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep dereplicate --debug --genomeInfo /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi/data/checkM/checkM_outdir/genome_info.csv -pa 0.99 -sa 0.999 -p 32 --S_algorithm fastANI --greedy_secondary_clustering /work_projet/mab_resistance/ANALYSIS/DREP_genomes_ncbi -g /work_projet/mab_resistance/DATA/GENOMES_NCBI/*.fna && conda deactivate"

```

**Résultats**

[dRep](https://github.com/vichitaok/projet_tutore/tree/main/results/ncbi_genomes/drep/figures)

NB: l'organisme du génome GCF_000611915.1_Mycobacterium_fortuitum_Z58_genomic a été renommé de "*Mycolicibacterium fortuitum* Z58" en "*Mycobacteroides abscessus* subsp. *abscessus*" en août 2018. [(https://www.ncbi.nlm.nih.gov/assembly/GCF_000611915.1)](https://www.ncbi.nlm.nih.gov/assembly/GCF_000611915.1).  
Ce qui explique   
1°) sa présence dans la liste des génomes de *M. abscessus* dans la recherche sur NCBI ;  
et 2°) la proximité de ce génome vis-à-vis des génomes de *M. abscessus* dans l'arbre.


## Séquences *M. abscessus* hôpital

* 81 séquences obtenues à partir de 81 souches isolées de prélèvements biologiques.
  
  - paired-end reads (2 x 150 pb)
  - Nextera XT DNA Library Preparation Kit / MiniSeq (300-cycles) et MiSeq 
  
  

### Contrôle qualité :

  [rapport MultiQC](https://vichitaok.github.io/projet_tutore/results/mab_hopital/multiqc_report.html)

```{bash fastqc}
#creation repertoires
for f in /work_projet/mab_resistance/DATA/MAB_HOPITAL/*.fastq.gz ; do mkdir -p /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/`basename $f .fastq.gz`/1-QC/fastqc/ ; done

#fastqc
for f in /work_projet/mab_resistance/DATA/MAB_HOPITAL/*.fastq.gz ; do 
  qsub -cwd -V -N fastqc-`basename ${f} .fastq.gz` 
  -q short.q -pe thread 4 
  -o /work_projet/mab_resistance/log/`basename ${f} .fastq.gz` 
  -e /work_projet/mab_resistance/log/`basename ${f} .fastq.gz` 
  -b y 
  "conda activate fastqc-0.11.8 && fastqc ${f} 
  --outdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/`basename ${f} .fastq.gz`/1-QC/fastqc/  
  && conda deactivate" ; 
done

#multiqc
mkdir -p /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/1-QC/multiqc/
mkdir -p /work_projet/mab_resistance/log/global/
qsub -cwd -V -N multiqc -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate multiqc-1.7 && multiqc -ip -o /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/1-QC/multiqc/ /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/1-QC/fastqc/ && conda deactivate"


```

### Nettoyage: outil fastp

```{bash fastp}
#liste des souches
ls /work_projet/mab_resistance/DATA/MAB_HOPITAL/ | sed 's/_R[12].fastq.gz//g' | uniq > /work_projet/mab_resistance/tables/mab_hopital_liste.txt

#fastp running
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do 
qsub -cwd -V -N fastp-${id} 
-pe thread 4 
-o /work_projet/mab_resistance/log/${id} 
-e /work_projet/mab_resistance/log/${id} 
-b y 
"conda activate fastp-0.20.0 && fastp 
--in1 /work_projet/mab_resistance/DATA/MAB_HOPITAL/${id}_R1.fastq.gz 
--in2 /work_projet/mab_resistance/DATA/MAB_HOPITAL/${id}_R1.fastq.gz 
--out1 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R1.cleaned_filtered.fastq.gz 
--out2 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R2.cleaned_filtered.fastq.gz 
--html /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_fastp.html 
--thread 4 
--detect_adapter_for_pe 
--cut_mean_quality 30 
--cut_window_size 8 
--length_required 100 
--cut_tail 
--json CLEANING/fastp.json && 
conda deactivate" ; 
done


#fastp en retirant les N bases
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do qsub -cwd -V -N fastp-${id} -pe thread 4 -o /work_projet/mab_resistance/log/${id} -e /work_projet/mab_resistance/log/${id} -b y "conda activate fastp-0.20.0 && fastp --in1 /work_projet/mab_resistance/DATA/MAB_HOPITAL/${id}_R1.fastq.gz --in2 /work_projet/mab_resistance/DATA/MAB_HOPITAL/${id}_R1.fastq.gz --out1 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R1.cleaned_filtered.fastq.gz --out2 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R2.cleaned_filtered.fastq.gz --html /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_fastp.html --thread 4 --detect_adapter_for_pe --cut_mean_quality 30 --cut_window_size 8 --length_required 100 --cut_tail --n_base_limit 0 --json CLEANING/fastp.json && conda deactivate" ; done


```


* Contrôle qualité après nettoyage:

  [rapport MultiQC après nettoyage](https://vichitaok.github.io/projet_tutore/results/mab_hopital/multiqc_on_cleaned_report.html)

```{bash}
#fastqc post-trimming
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do qsub -cwd -V -N fastqc-cleaned-${id} -q short.q -pe thread 4 -o /work_projet/mab_resistance/log/${id}/ -e /work_projet/mab_resistance/log/${id}/ -b y "conda activate fastqc-0.11.8 && fastqc /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/*.fastq.gz --outdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/1-QC/fastqc/  && conda deactivate" ; done


## multiqc global
# répertoire pour résultat
mkdir -p /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/1-QC/multiqc-cleaned-fastp/

# multiqc running
qsub -cwd -V -N multiqc -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate multiqc-1.7 && multiqc -ip -o /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/1-QC/multiqc-cleaned-fastp/ /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/1-QC/fastqc/ /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/2-CLEANING/fastp/ && conda deactivate"

```


[rapport global](https://vichitaok.github.io/projet_tutore/results/mab_hopital/multiqc_on_cleaned_fastp_report.html)


### Statistiques sur les données: 

```{bash seqkit stats}
qsub -cwd -V -N seqkit-stats -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate seqkit-0.11.0 && seqkit stats /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/2-CLEANING/fastp/*.cleaned_filtered.fastq.gz > /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/seqkit_stats_on_cleaned.txt && conda deactivate"

```

```{r libraries, eval=TRUE}
#### Required libraries ####

# Load required CRAN R libraries
required_cranLib <- c("knitr", 
                      "tidyverse")  # data manipulation
for (lib in required_cranLib) {
  if (!require(lib, character.only = TRUE)) {
    install.packages(lib)
  }
  require(lib, character.only = TRUE)
}

kable(as.data.frame(c(required_cranLib)),
      col.names = "libraries",
      caption = "Loaded required libraries"
    )
```


```{r calcul profondeurs, eval=TRUE}
# nombre décimales
options(digits=3)

# load tableau résultats seqkit
reads_stat <- read.table("/work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/seqkit_stats_on_cleaned.txt",
                         header = TRUE)

# renommer les noms des reads (1ère colonne file)
reads_stat$file <- str_sub(reads_stat$file, 73, 77)

# transformer la colonne num_seqs (chr) en numeric pour calculs
reads_stat$num_seqs_num <- as.numeric(str_replace_all(reads_stat$num_seqs, ",", ""))

# calcul profondeur à partir des nb de sequences et longueur moyenne des reads
reads_stat <- reads_stat %>%
  mutate(depth = (as.numeric(num_seqs_num) * avg_len) / 2.5e6) # taille génome M.asbcessus ~ 5 Mbp

# supprimer 1 des 2 lignes de chaque échantillon
reads_stat <- reads_stat %>%
  unique()

# noms des lignes = noms des échantillons
reads_stat_depth <- reads_stat %>% 
  remove_rownames %>% 
  column_to_rownames(var = "file")

# view (reads_stat_depth)
kable(reads_stat_depth,
      caption = "Seqkit stats reads M.abscessus")

# summary
summary(reads_stat_depth)
```


La qualité des reads semble bonne glogalement. Avec un pourcentage en GC de **64 %**, cohérent avec l'espèce *M.abscessus*.  

A noter la longueur variable des reads, en fonction des échantillons ; évoquant un filtrage préalable sur les reads.

Par ailleurs, les échantillons **mab06** et **mab52** présentent des tailles de fichiers, des longueurs de reads bien inférieures. La profondeur est de 18 et 17, respectivement. Ce qui rendra difficile l'assemblage *de novo*.


*Verification contamination mab06 et mab52

```{bash dRep comparaison genomes prives ncbi greedy_secondary_clustering + fast_ANI}
qsub -cwd -V -N drep-mab-hopital-097 -pe thread 32 -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -q long.q -b y "conda activate drep-3.2.0 && dRep dereplicate -d -pa 0.97 -p 32 --S_algorithm fastANI --greedy_secondary_clustering /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/DREP_mab_hopital_pa_0.97 --genomes /work_projet/mab_resistance/tables/path_mab_hopital.txt && conda deactivate"

 qsub -cwd -V -N drep-all-genomes-097 -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare --debug -pa 0.97 -p 32 --S_algorithm fastANI /work_projet/mab_resistance/ANALYSIS/DREP_all_genomes_pa_0.97 --genomes /work_projet/mab_resistance/tables/path_to_all_genomes.txt && conda deactivate"

```

mab06 : contamination 0
mab52 : contamination 0.46

### Assemblage : 


test assemblage unicycler sur mab01
81 contigs
- Spades (Unicycler)
```{bash unicycler test}
# création répertoire assembly
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do mkdir -p /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/ ; done

## Test sur un échantillon mab01
qsub -cwd -V -N unicycler -pe thread 4  -o /work_projet/mab_resistance/log/mab01/ -e /work_projet/mab_resistance/log/mab01/ -b y "conda activate unicycler-0.4.8 && unicycler  --short1 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/2-CLEANING/fastp/mab01_R1.cleaned_filtered.fastq.gz  --short2 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/2-CLEANING/fastp/mab01_R2.cleaned_filtered.fastq.gz   --out /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/3-Assembly/ --threads 4 && conda deactivate" 


```

- Qualité de l'assemblage: Quast
```{bash quast}
qsub -cwd -V -N quast -pe thread 4 -o /work_projet/mab_resistance/log/mab01/ -e /work_projet/mab_resistance/log/mab01/  -b y " conda activate  quast-5.0.2 && quast.py  -o /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/3-Assembly/quast_results/ /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/3-Assembly/mab01.fasta  && conda deactivate"
```




* Unicycler sur l'ensemble des génomes
```{bash unicycler}
# Running Unicycler
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do 
qsub -cwd -V -N unicycler-${id} -pe thread 4  
-o /work_projet/mab_resistance/log/${id}/ 
-e /work_projet/mab_resistance/log/${id}/ 
-b y "conda activate unicycler-0.4.8 && unicycler  
--short1 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R1.cleaned_filtered.fastq.gz  
--short2 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/2-CLEANING/fastp/${id}_R2.cleaned_filtered.fastq.gz   
--out /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/ 
--threads 4 
&& conda deactivate" ; done

# Renommer les final assemblies
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt`; do echo ${id} | mv /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/assembly.fasta /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/${id}.fasta ; done


```

* Quast sur l'ensemble des génomes, avec un génome de référence (celui de Mycobrowser, *i.e.* *Mycobacterium abscessus* ATCC19977)  

NB: Les génome **GCF_001606295.1 (Mycobacteroides abscessus ASM160629v1 RefSeq)** et **Mycobacterium abscessus ATCC19977** ne figurent pas dans la liste des génomes téléchargés, qui ne comprennent que la sous-espèce *M.abscessus* subsp. *abscessus*.

Par ailleurs, le génome de référence de Mycobrowser *Mycobacterium abscessus* ATCC19977 est un génome public de NCBI (NC_010397.1), lequel a été retiré par l'équipe RefSeq [(https://www.ncbi.nlm.nih.gov/nuccore/NC_010397.1?report=genbank)](https://www.ncbi.nlm.nih.gov/nuccore/NC_010397.1?report=genbank).
Probablement en raison d'une redondance de génomes publiés, avec CU458896, auquel la même référence Pubmed est associée.  
Ripoll F *et al*. Non mycobacterial virulence genes in the genome of the emerging pathogen *Mycobacterium abscessus*. POne 2009.

En commentaire du Genbank, on observe en effet :

```
COMMENT     REVIEWED REFSEQ: This record has been curated by NCBI staff. The
            reference sequence is identical to CU458896.
```
Ce génome ATCC19977 (CU458896, GCA_000069185) ne fait pas partie des génomes téléchargés car étiquetée *M.abscessus* et non sous-espèce *M.abscessus* subsp. *abscessus*.

```{bash quast avec un genome de ref}

mkdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/3-ASSEMBLY/

qsub -cwd -V -N quast_global_atcc -pe thread 4  -o /work_projet/mab_resistance/log/global/  -e /work_projet/mab_resistance/log/global/  -b y "conda activate  quast-5.0.2 && quast.py -o /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/3-ASSEMBLY/quast_results/ -r /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_ATCC_19977_genome_v4.fasta /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/3-ASSEMBLY/*.fasta --threads 4 && conda deactivate"


mkdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/multiqc-assembly

qsub -cwd -V -N multiqc-assembly -pe thread 4 -o /work_projet/mab_resistance/log/global/  -e /work_projet/mab_resistance/log/global/ -b y "conda activate multiqc-1.7 && multiqc -o /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/multiqc-assembly/  /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/3-ASSEMBLY/quast_results/  && conda deactivate"

```

Résultats :
[Rapport MultiQC Assemblage](https://vichitaok.github.io/projet_tutore/results/mab_hopital/multiqc_assembly_report.html)

Comme attendu, l'assemblage des reads des échantillons de *mab06* et *mab52* ne donne pas de résultats satisfaisants. Avec des N50 de 15.8 et 3.4 kbp, respectivement et des contigs de petite taille (max 67.9 kbp et 48.2 kbp).

Par ailleurs, pour *mb02* : on observe des mismatchs avec missassemblies. 

```{bash dRep compare sur ensemble des génomes}
mkdir /work_projet/mab_resistance/ANALYSIS/DREP_all_genomes

qsub -cwd -V -N drep-all-genomes -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare --debug -pa 0.99 -sa 0.999 -p 32 --S_algorithm fastANI --greedy_secondary_clustering /work_projet/mab_resistance/ANALYSIS/DREP_all_genomes --genomes /work_projet/mab_resistance/tables/path_to_genomes.txt && conda deactivate"
$ error : ne trouve pas les inputs
```

mais liste à refaire (chemins complets)

```{bash liste des chemins}
cd /work_projet/mab_resistance/DATA/GENOMES_NCBI/
ls -d $PWD/* > /work_projet/mab_resistance/tables/path_to_genomes.txt

echo /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_ATCC_19977_genome_v4.fasta >> /work_projet/mab_resistance/tables/path_to_genomes.txt

for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt`; do echo /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/${id}.fasta >> /work_projet/mab_resistance/tables/path_to_genomes.txt ; done

# contrôle nombre de lignes 
[vok@migale GENOMES_NCBI]$ cat /work_projet/mab_resistance/tables/path_to_genomes.txt | wc -l
994



# running dRep
qsub -cwd -V -N drep-all-genomes -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare --debug -pa 0.99 -sa 0.999 -p 32 --S_algorithm fastANI --greedy_secondary_clustering /work_projet/mab_resistance/ANALYSIS/DREP_all_genomes --genomes /work_projet/mab_resistance/tables/path_to_genomes.txt && conda deactivate"

# running dRep pa 0.97 greedy_secondary
qsub -cwd -V -N drep-all-genomes-097 -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare --debug -pa 0.97 -p 32 --S_algorithm fastANI --greedy_secondary_clustering /work_projet/mab_resistance/ANALYSIS/DREP_all_genomes_pa_0.97 --genomes /work_projet/mab_resistance/tables/path_to_genomes.txt && conda deactivate"

# running dRep pa 0.97
qsub -cwd -V -N drep-all-genomes-097 -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare --debug -pa 0.97 -p 32 --S_algorithm fastANI /work_projet/mab_resistance/ANALYSIS/DREP_all_genomes_pa_0.97 --genomes /work_projet/mab_resistance/tables/path_to_genomes.txt && conda deactivate"

# changement nom du répertoire en DREP_all_genomes_1atcc_pa_0.97
```

```{bash ajout de 2 genomes "de référence"}
# ajout de 2 genomes "de référence", le NZ_CP014955.1 et le CU458896.1
cd /work_projet/mab_resistance/DATA/GENOMES_NCBI/
ls -d $PWD/* > /work_projet/mab_resistance/tables/path_to_all_genomes.txt

echo /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_ATCC_19977_genome_v4.fasta >> /work_projet/mab_resistance/tables/path_to_all_genomes.txt

echo /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_atcc_19977_CU458896-1_genome.fasta >> /work_projet/mab_resistance/tables/path_to_all_genomes.txt
echo /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_NZ_CP014955-1_genome.fasta >> /work_projet/mab_resistance/tables/path_to_all_genomes.txt

for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt`; do echo /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/${id}.fasta >> /work_projet/mab_resistance/tables/path_to_all_genomes.txt ; done


# contrôle nombre de lignes 
[vok@migale GENOMES_NCBI]$ cat /work_projet/mab_resistance/tables/path_to_all_genomes.txt | wc -l
996



# running dRep pa 0.97
qsub -cwd -V -N drep-all-genomes-097 -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare --debug -pa 0.97 -p 32 --S_algorithm fastANI /work_projet/mab_resistance/ANALYSIS/DREP_all_genomes_pa_0.97 --genomes /work_projet/mab_resistance/tables/path_to_all_genomes.txt && conda deactivate"

# running dRep pa 0.99
qsub -cwd -V -N drep-all-genomes-099 -pe thread 32 -o /work_projet/mab_resistance/log/ -e /work_projet/mab_resistance/log/ -q long.q -b y "conda activate drep-3.2.0 && dRep compare --debug -pa 0.99 -p 32 --S_algorithm fastANI /work_projet/mab_resistance/ANALYSIS/drep_compare_all_genomes_pa_0.99 --genomes /work_projet/mab_resistance/tables/path_to_all_genomes.txt && conda deactivate"

```


```{bash checkM mab hopital}
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt`; do echo /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/${id}.fasta >> /work_projet/mab_resistance/tables/path_mab_hopital.txt ; done

$ cat /work_projet/mab_resistance/tables/path_mab_hopital.txt | wc -l
81


qsub -cwd -V -N checkm-lineage-mab_hopital -pe thread 32 -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -q long.q -b y "conda activate drep-3.2.0 && checkm lineage_wf /work_projet/mab_resistance/ /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/DREP_mab_hopital/data/checkM/checkM_outdir/ -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/DREP_mab_hopital/data/checkM/checkM_outdir/results.tsv --tab_table -t 32 --pplacer_threads 32 -g -x faa --tmpdir /work_projet/mab_resistance/tmp/ && conda deactivate"

qsub -cwd -V -N checkm-qa-mab_hopital -pe thread 32 -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -q long.q -b y "conda activate drep-3.2.0 && checkm qa /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/DREP_mab_hopital/data/checkM/checkM_outdir/lineage.ms /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/DREP_mab_hopital/data/checkM/checkM_outdir/ -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/DREP_mab_hopital/data/checkM/checkM_outdir/Chdb.tsv -t 32 --tab_table -o 2 --tmpdir /work_projet/mab_resistance/tmp/ && conda deactivate"

# checkm mab06 et 52
qsub -cwd -V -N checkm-lineage-mab06 -pe thread 8 -o /work_projet/mab_resistance/log/mab06/ -e /work_projet/mab_resistance/log/mab06/ -b y "conda activate drep-3.2.0 && checkm lineage_wf /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab06/3-ASSEMBLY/mab06.fasta /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab06/checkm/data/checkM/checkM_outdir/ -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab06/checkm/data/checkM/checkM_outdir/results.tsv --tab_table -t 8 --pplacer_threads 8 -g -x faa --tmpdir /work_projet/mab_resistance/tmp/ && conda deactivate"

```

## Extraction séquences des 35 gènes de la liste

```{bash }
# creer liste des id des genes de resistance tels qu'ils apparaissent dans le multifasta
for id in `cat /work_projet/mab_resistance/tables/liste_genes_resistance.txt` ; do grep ${id} /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_ATCC_19977_genes_v4.fasta | sed 's/^>//' >> /work_projet/mab_resistance/tables/liste_id_resistance.txt ; done

```


```{python multifasta des 35 genes}
# selection des 35 fasta correspondant à ces gènes
# script python (SeqIO)
# BLAST de ce nouveau multifasta pour avoir les id Genbank
```



```{bash blast multifasta genes resistance}
qsub -cwd -V -N blast-multifasta-NZ_CP014955 -pe thread 16 -o /work_projet/mab_resistance/log/reference/ -e /work_projet/mab_resistance/log/reference/ -b y "conda activate blast-2.9.0 && blastn -query /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/attc_resistance.fasta -subject  /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_NZ_CP014955-1_genome.fasta -num_threads 16 -outfmt 6 -out /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/attc_resistance_blast_NZ_CP014955-1.tsv && conda deactivate"

#sur  multifasta
qsub -cwd -V -N blast-multifasta-NZ_CP014955 -pe thread 16 -o /work_projet/mab_resistance/log/reference/ -e /work_projet/mab_resistance/log/reference/ -b y "conda activate blast-2.9.0 && blastn -query /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/attc_resistance.fasta -subject  /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_NZ_CP014955-1_coding_sequences.fasta -num_threads 16 -outfmt 6 -out /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/attc_resistance_blast_NZ_CP014955-1.tsv && conda deactivate"


#sur  multifasta CU458896
qsub -cwd -V -N blast-multifasta-CU458896 -pe thread 16 -o /work_projet/mab_resistance/log/reference/ -e /work_projet/mab_resistance/log/reference/ -b y "conda activate blast-2.9.0 && blastn -query /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/genes_resistance_atcc.fasta -subject  /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_atcc_19977_CU458896-1_coding_sequences.fasta -num_threads 16 -outfmt 6 -out /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/attc_resistance_blast_atcc_19977_CU458896-1.tsv && conda deactivate"

```


 
```{r, eval=TRUE}


atcc_blast_refseq <- read.table("/work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/atcc_resistance_blast_NZ_CP014955-1.tsv", header = FALSE)

names(atcc_blast_refseq) <- c("qseqid", "sseqid", "pident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")

atcc_blast_refseq_sort <- atcc_blast_refseq %>%
  arrange(desc(pident), qseqid)

atcc_blast_refseq_filter <- atcc_blast_refseq %>%
  filter(pident > 99.5) %>%
  filter(pident != 	99.907)  # enlever le deuxième 	MAB_4099c

write.table(atcc_blast_refseq_filter, file = "/work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/atcc_resistance_blast_NZ_CP014955-1_filter.tsv", sep = "\t")


```

 
```{r}
atcc_blast_CU458896 <- read.table("/work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/atcc_resistance_blast_atcc_19977_CU458896-1.tsv", header = FALSE)

names(atcc_blast_CU458896) <- c("qseqid", "sseqid", "pident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")

atcc_blast_CU458896_sort <- atcc_blast_CU458896 %>%
  arrange(desc(pident), qseqid)

atcc_blast_CU458896_filter <- atcc_blast_CU458896 %>%
  filter(pident > 99.5)%>%
  filter(length != 	166)  # enlever le deuxième 	MAB_4395

write.table(atcc_blast_CU458896_filter, file = "/work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/atcc_resistance_blast_atcc_19977_CU458896-1_filter.tsv", sep = "\t")


```


```{bash}
cd /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/
[vok@migale genes_resistance]$ sed 's/"//g' atcc_resistance_blast_NZ_CP014955-1_filter.tsv | sed '1d' | cut -f 3 | cut -f 4,5 -d "_" > liste_id_res_NZ_CP014955-1.txt
[vok@migale genes_resistance]$ cat liste_id_res_NZ_CP014955-1.txt | wc -l
33

cd /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/
[vok@migale genes_resistance]$ sed 's/"//g' atcc_resistance_blast_atcc_19977_CU458896-1_filter.tsv | sed '1d' | cut -f 3 | cut -f 3 -d "_" > liste_id_res_CU458896-1.txt
[vok@migale genes_resistance]$ cat liste_id_res_CU458896-1.txt | wc -l
33
```


```{bash}
# test avec fichier annotation gb Mycobacterium_abscessus_NZ_CP014955-1 (avec sequence)
qsub -cwd -V -N prokka-mab01-NZ_CP014955-gb-seq -pe thread 8  -o /work_projet/mab_resistance/log/test/ -e /work_projet/mab_resistance/log/test/ -b y "conda activate prokka-1.14.5 && prokka --outdir /work_projet/mab_resistance/test/mab01_NZ_CP014955_gb_seq --prefix mab01-test --locustag mab01-test --gcode 11 --proteins /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_NZ_CP014955-1.gb --cpus 8 --rnammer /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/3-ASSEMBLY/mab01.fasta && conda deactivate"
```

## Annotation

Prokka sur l'ensemble des génomes
```{bash prokka tout les génomes mab hopital}
# Running Prokka NZ_CP014955_gb
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do 
qsub -cwd -V -N prokka-${id} -pe thread 8  
-o /work_projet/mab_resistance/log/${id}/ 
-e /work_projet/mab_resistance/log/${id}/ 
-b y "conda activate prokka-1.14.5 && prokka --outdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/4-ANNOTATION/prokka_${id}_NZ_CP014955_gb/ --prefix ${id} --locustag ${id} --gcode 11 --proteins /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_NZ_CP014955-1.gb --cpus 8 --rnammer /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/${id}.fasta && conda deactivate" ; done


# Running Prokka CU458896_gb
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do 
qsub -cwd -V -N prokka-${id}-CU458896 -pe thread 8  
-o /work_projet/mab_resistance/log/${id}/ 
-e /work_projet/mab_resistance/log/${id}/ 
-b y "conda activate prokka-1.14.5 && prokka --outdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/4-ANNOTATION/prokka_${id}_CU458896_gb/ --prefix ${id} --locustag ${id} --gcode 11 --proteins /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_atcc_19977_CU458896-1.gb --cpus 8 --rnammer /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/${id}.fasta && conda deactivate" ; done

for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do qsub -cwd -V -N prokka-${id}-CU458896 -pe thread 8 -o /work_projet/mab_resistance/log/${id}/ -e /work_projet/mab_resistance/log/${id}/ -b y "conda activate prokka-1.14.5 && prokka --outdir /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/4-ANNOTATION/prokka_${id}_CU458896_gb/ --prefix ${id} --locustag ${id} --gcode 11 --proteins /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_atcc_19977_CU458896-1.gb --cpus 8 --rnammer /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/3-ASSEMBLY/${id}.fasta && conda deactivate" ; done

```


# Pan-génome


```{bash roary NZ}
# test
qsub -cwd -V -N roary-test -pe thread 8  -o /work_projet/mab_resistance/log/test/ -e /work_projet/mab_resistance/log/test/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/4-ANNOTATION/prokka_mab01_CU458896_gb/mab01.gff -f /work_projet/mab_resistance/ANALYSIS/test/roary_mab01_test_NZ_CP014955_gb/ && conda deactivate"

cd /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/mab01/4-ANNOTATION/prokka_mab01_CU458896_gb/
qsub -cwd -V -N roary-test -pe thread 8  -o /work_projet/mab_resistance/log/test/ -e /work_projet/mab_resistance/log/test/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 *.gff -f /work_projet/mab_resistance/ANALYSIS/test/roary_mab01_test_CU458896_gb/ && conda deactivate"



# Running Roary _NZ_CP014955_gb
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do 
qsub -cwd -V -N roary-${id}-NZ_CP014955 -pe thread 8  
-o /work_projet/mab_resistance/log/${id}/ 
-e /work_projet/mab_resistance/log/${id}/ 
-b y "conda activate roary-3.12.0 && roary –e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/4-ANNOTATION/prokka_${id}_NZ_CP014955_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/5-PANGENOME/roary_${id}_NZ_CP014955_gb/ && conda deactivate" ; done


for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do qsub -cwd -V -N roary-${id}-NZ_CP014955 -pe thread 8  -o /work_projet/mab_resistance/log/${id}/ -e /work_projet/mab_resistance/log/${id}/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/4-ANNOTATION/prokka_${id}_NZ_CP014955_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/5-PANGENOME/roary_${id}_NZ_CP014955_gb/ && conda deactivate" ; done

# Running Roary CU458896_gb
for id in `cat /work_projet/mab_resistance/tables/mab_hopital_liste.txt` ; do 
qsub -cwd -V -N roary-${id}-CU458896 -pe thread 8  
-o /work_projet/mab_resistance/log/${id}/ 
-e /work_projet/mab_resistance/log/${id}/ 
-b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/4-ANNOTATION/prokka_${id}_CU458896_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/${id}/5-PANGENOME/roary_${id}_CU458896_gb/ && conda deactivate" ; done

qsub -cwd -V -N roary-CU458896 -pe thread 8  -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/4-ANNOTATION/prokka_*_CU458896_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/4-PANGENOME/roary_CU458896/ && conda deactivate"

# NZ_CP014955
qsub -cwd -V -N roary-NZ_CP014955 -pe thread 8  -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/4-ANNOTATION/prokka_*_NZ_CP014955_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/4-PANGENOME/roary_NZ_CP014955/ && conda deactivate"

```

```{bash roary sans les mab06 et 52}
# Running Roary NZ_CP014955
qsub -cwd -V -N roary-NZ_CP014955-clean -pe thread 8  -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/4-ANNOTATION/prokka_*_NZ_CP014955_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/4-PANGENOME/roary_NZ_CP014955_cleaned/ && conda deactivate"

# Running Roary CU458896_gb
qsub -cwd -V -N roary-CU458896-clean -pe thread 8  -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/4-ANNOTATION/prokka_*_CU458896_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/global/4-PANGENOME/roary_CU458896_cleaned/ && conda deactivate"

```


```{bash extraction genes avec id}
for geneid in `cat /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/liste_id_res_CU458896-1.txt` ; do echo ${geneid} && grep ${geneid} /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_atcc_19977_CU458896-1.gff3 > /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance_CU458896-1/${geneid}.gff ; done
	
for geneid in `cat /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/liste_id_res_NZ_CP014955-1.txt` ; do echo ${geneid} && grep ${geneid} /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_NZ_CP014955-1.gff3 > /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance_NZ_CP014955-1/${geneid}.gff ; done
	
	
for geneid in `cat /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/liste_id_res_CU458896-1.txt` ; do qsub -cwd -V -N getfasta-CU458896-${geneid} -q short.q -pe thread 4 -o /work_projet/mab_resistance/log/reference/ -e /work_projet/mab_resistance/log/reference/ -b y "conda activate bedtools-2.29.0 && bedtools getfasta -fi /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_atcc_19977_CU458896-1_genome.fasta -bed /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance_CU458896-1/${geneid}.gff -fo /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance_CU458896-1/${geneid}.fasta  && conda deactivate" ; done	


for geneid in `cat /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance/liste_id_res_NZ_CP014955-1.txt` ; do qsub -cwd -V -N getfasta-NZ_CP014955-${geneid} -q short.q -pe thread 4 -o /work_projet/mab_resistance/log/reference/ -e /work_projet/mab_resistance/log/reference/ -b y "conda activate bedtools-2.29.0 && bedtools getfasta -fi /work_projet/mab_resistance/DATA/REFERENCE/Mycobacterium_abscessus_NZ_CP014955-1_genome.fasta -bed /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance_NZ_CP014955-1/${geneid}.gff -fo /work_projet/mab_resistance/ANALYSIS/REFERENCE/genes_resistance_NZ_CP014955-1/${geneid}.fasta  && conda deactivate" ; done



```


* Téléchargement des genbank des winner genomes
```{bash telechargement winner genomes}
# liste des winner genomes
ls /work_projet/mab_resistance/ANALYSIS/GENOMES_NCBI/DREP_genomes_ncbi/dereplicated_genomes/ | sed 's/_genomic.fna//g' > /work_projet/mab_resistance/tables/ncbi_winner_genomes.txt

# soumission du script pour telecharger les gff
qsub -cwd -V -N ncbi_winner_genomes_gff -o /work_projet/mab_resistance/log/genomes_ncbi -e /work_projet/mab_resistance/log/genomes_ncbi /save_projet/mab_resistance/projet_tutore/scripts/ncbi_winner_genomes_gff.sh


# soumission du script pour telecharger les gbff
qsub -cwd -V -N ncbi_winner_genomes_gbff -o /work_projet/mab_resistance/log/genomes_ncbi -e /work_projet/mab_resistance/log/genomes_ncbi /save_projet/mab_resistance/projet_tutore/scripts/ncbi_winner_genomes_gbff.sh

# création répertoire pour les gff prets pour roary
mkdir /work_projet/mab_resistance/ANALYSIS/pangenome
mkdir /work_projet/mab_resistance/ANALYSIS/pangenome/gff_NZ_CP014955_gb
mkdir /work_projet/mab_resistance/ANALYSIS/pangenome/gff_CU458896_gb
mkdir /work_projet/mab_resistance/ANALYSIS/pangenome/gff_ncbi
mkdir /work_projet/mab_resistance/ANALYSIS/pangenome/gff_ref

# transformer les .gbff en gff3 avec bp_genbank2gff3.pl
qsub -cwd -V -N bp_genbank2gff3 -o /work_projet/mab_resistance/log/genomes_ncbi -e /work_projet/mab_resistance/log/genomes_ncbi 

bp_genbank2gff3.pl /work_projet/mab_resistance/DATA/GENOMES_NCBI/gbff/*gbff.gz --outdir /work_projet/mab_resistance/ANALYSIS/pangenome/gff_ncbi/ 1>/work_projet/mab_resistance/log/genomes_ncbi/bp_genbank2gff3_o.log 2>/work_projet/mab_resistance/log/genomes_ncbi/bp_genbank2gff3_e.log

bp_genbank2gff3.pl /work_projet/mab_resistance/DATA/REFERENCE/*.gb --outdir /work_projet/mab_resistance/ANALYSIS/pangenome/gff_ref/ 1>/work_projet/mab_resistance/log/genomes_ncbi/bp_genbank2gff3_ref_o.log 2>/work_projet/mab_resistance/log/genomes_ncbi/bp_genbank2gff3_ref_e.log

# copie des gff
#cp /work_projet/mab_resistance/ANALYSIS/pangenome/gff_NZ_CP014955_gb/*.gff3 /work_projet/mab_resistance/ANALYSIS/pangenome/gff_CU458896_gb/

cp /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/4-ANNOTATION/prokka_*_NZ_CP014955_gb/*.gff /work_projet/mab_resistance/ANALYSIS/pangenome/gff_NZ_CP014955_gb/

cp /work_projet/mab_resistance/ANALYSIS/MAB_HOPITAL/*/4-ANNOTATION/prokka_*_CU458896_gb/*.gff /work_projet/mab_resistance/ANALYSIS/pangenome/gff_CU458896_gb/

```


* Roary sans les mab06, 52 et 58
```{bash roary mab hopital sans les mab06, 52 et 58}
# Running Roary NZ_CP014955
qsub -cwd -V -N roary-NZ_CP014955 -pe thread 8  -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/pangenome/gff_NZ_CP014955_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/pangenome/roary_NZ_CP014955_mab_hopital_cleaned/ && conda deactivate"

# Running Roary CU458896_gb
qsub -cwd -V -N roary-CU458896 -pe thread 8  -o /work_projet/mab_resistance/log/global/ -e /work_projet/mab_resistance/log/global/ -b y "conda activate roary-3.12.0 && roary -e --mafft -p 8 /work_projet/mab_resistance/ANALYSIS/pangenome/gff_CU458896_gb/*.gff -f /work_projet/mab_resistance/ANALYSIS/pangenome/roary_CU458896_mab_hopital_cleaned/ && conda deactivate"

```

